nnmnkwii/paramgen/_mlpg.py

```diff
@@ -334,14 +349,31 @@ def unit_variance_mlpg_matrix(windows, T):
     """
     win_mats = build_win_mats(windows, T)
     sdw = np.max([win_mat.l + win_mat.u for win_mat in win_mats])
+    max_win_width = np.max([max(win_mat.l, win_mat.u) for win_mat in win_mats])
 
     P = bm.zeros(sdw, sdw, T)
+
+    # set edge precitions to zero
+    precisions = bm.zeros(0, 0, T)
+    precisions.data[:, max_win_width:-max_win_width] += 1.0
+
+    mod_win_mats = list()
     for win_index, win_mat in enumerate(win_mats):
-        bm.dot_mm_plus_equals(win_mat.T, win_mat, target_bm=P)
+        if win_index != 0:
+            # use zero precisions for dynamic features
+            mod_win_mat = bm.dot_mm(precisions, win_mat)
+            bm.dot_mm_plus_equals(mod_win_mat.T, win_mat, target_bm=P)
```

Comment:
Did you mean 

```py
bm.dot_mm_plus_equals(mod_win_mat.T, mod_win_mat, target_bm=P)
```

?

(probably `mod_win_mat.T.dot(win_mat)` = `mod_win_mat.T.dot(mod_win_mat)`, but let me double-check.

Reply:
Thank you for your comment.
Either is OK in this case.

Theoretically, the method calculates
<a href="https://www.codecogs.com/eqnedit.php?latex=R&space;=&space;\left(&space;&\begin{bmatrix}&space;W_1^T&space;&&space;W_2^T&space;&&space;W_3^T&space;\end{bmatrix}&space;\begin{bmatrix}&space;P_1&space;&&space;&&space;\\&space;&&space;P_2&space;&&space;\\&space;&&space;&&space;P_3&space;\end{bmatrix}&space;\begin{bmatrix}&space;W_1&space;\\&space;W_2&space;\\&space;W_3&space;\end{bmatrix}&space;\right)^{-1}&space;\begin{bmatrix}&space;W_1^T&space;&&space;W_2^T&space;&&space;W_3^T&space;\end{bmatrix}&space;\begin{bmatrix}&space;P_1&space;&&space;&&space;\\&space;&&space;P_2&space;&&space;\\&space;&&space;&&space;P_3&space;\end{bmatrix}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?R&space;=&space;\left(&space;&\begin{bmatrix}&space;W_1^T&space;&&space;W_2^T&space;&&space;W_3^T&space;\end{bmatrix}&space;\begin{bmatrix}&space;P_1&space;&&space;&&space;\\&space;&&space;P_2&space;&&space;\\&space;&&space;&&space;P_3&space;\end{bmatrix}&space;\begin{bmatrix}&space;W_1&space;\\&space;W_2&space;\\&space;W_3&space;\end{bmatrix}&space;\right)^{-1}&space;\begin{bmatrix}&space;W_1^T&space;&&space;W_2^T&space;&&space;W_3^T&space;\end{bmatrix}&space;\begin{bmatrix}&space;P_1&space;&&space;&&space;\\&space;&&space;P_2&space;&&space;\\&space;&&space;&&space;P_3&space;\end{bmatrix}" title="R = \left( &\begin{bmatrix} W_1^T & W_2^T & W_3^T \end{bmatrix} \begin{bmatrix} P_1 & & \\ & P_2 & \\ & & P_3 \end{bmatrix} \begin{bmatrix} W_1 \\ W_2 \\ W_3 \end{bmatrix} \right)^{-1} \begin{bmatrix} W_1^T & W_2^T & W_3^T \end{bmatrix} \begin{bmatrix} P_1 & & \\ & P_2 & \\ & & P_3 \end{bmatrix}" /></a>
where W_l and P_l are window and precision matrices, respectively.
Let <a href="https://www.codecogs.com/eqnedit.php?latex=\tilde{W_l}&space;=&space;P_lW_l" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\tilde{W_l}&space;=&space;P_lW_l" title="\tilde{W_l} = P_lW_l" /></a>, we obtain
<a href="https://www.codecogs.com/eqnedit.php?latex=R&space;=&space;\left(&space;\begin{bmatrix}&space;\tilde{W}_1^T&space;&&space;\tilde{W}_2^T&space;&&space;\tilde{W}_3^T&space;\end{bmatrix}&space;\begin{bmatrix}&space;W_1&space;\\&space;W_2&space;\\&space;W_3&space;\end{bmatrix}&space;\right)^{-1}&space;\begin{bmatrix}&space;\tilde{W}_1^T&space;&&space;\tilde{W}_2^T&space;&&space;\tilde{W}_3^T&space;\end{bmatrix}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?R&space;=&space;\left(&space;\begin{bmatrix}&space;\tilde{W}_1^T&space;&&space;\tilde{W}_2^T&space;&&space;\tilde{W}_3^T&space;\end{bmatrix}&space;\begin{bmatrix}&space;W_1&space;\\&space;W_2&space;\\&space;W_3&space;\end{bmatrix}&space;\right)^{-1}&space;\begin{bmatrix}&space;\tilde{W}_1^T&space;&&space;\tilde{W}_2^T&space;&&space;\tilde{W}_3^T&space;\end{bmatrix}" title="R = \left( \begin{bmatrix} \tilde{W}_1^T & \tilde{W}_2^T & \tilde{W}_3^T \end{bmatrix} \begin{bmatrix} W_1 \\ W_2 \\ W_3 \end{bmatrix} \right)^{-1} \begin{bmatrix} \tilde{W}_1^T & \tilde{W}_2^T & \tilde{W}_3^T \end{bmatrix}" /></a>

The code comes from the equation.

In this case, since the precision matrix is diagonal and the elements are 0 or 1, they are equivalent as follows
<a href="https://www.codecogs.com/eqnedit.php?latex=\tilde{W}_l^T&space;\tilde{W}_l^T&space;=&space;W_l^T&space;P_l^\top&space;P_l&space;W_l&space;=&space;W_l^T&space;P_l^\top&space;W_l&space;=&space;\tilde{W}_l^T&space;W_l" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\tilde{W}_l^T&space;\tilde{W}_l^T&space;=&space;W_l^T&space;P_l^\top&space;P_l&space;W_l&space;=&space;W_l^T&space;P_l^\top&space;W_l&space;=&space;\tilde{W}_l^T&space;W_l" title="\tilde{W}_l^T \tilde{W}_l^T = W_l^T P_l^\top P_l W_l = W_l^T P_l^\top W_l = \tilde{W}_l^T W_l" /></a>

Reply:
I see. Thank you for the clarification!